docs/guide/widgets.md

```diff
@@ -388,3 +384,191 @@ The following builtin widgets use the Line API. If you are building advanced wid
+
+In this section we will show how to design and build a fully working app, while keeping widgets re-usable.
+
+### Designing the app
+
+We are going to build a *byte editor* which allows you to enter a number in both decimal and binary. You could use this a teaching aid for binary numbers.
+
+Here's a sketch of what the app should ultimately look like:
+
+!!! tip
+
+    There are plenty of resources on the web, such as this [excellent video from Khan Academy](https://www.khanacademy.org/math/algebra-home/alg-intro-to-algebra/algebra-alternate-number-bases/v/number-systems-introduction) if you want to brush up on binary numbers.
+
+
+<div class="excalidraw">
+--8<-- "docs/images/byte01.excalidraw.svg"
+</div>
+
+There are three types of built-in widget in the sketch, namely ([Input](../widgets/input.md), [Label](../widgets/label.md), and [Switch](../widgets/switch.md)). Rather than manage these as a single collection of widgets, we can arrange them in to logical groups with compound widgets. This will make our app easier to work with.
+
+###  Identifying components
+
+We will divide this UI in to three compound widgets:
+
+1. `BitSwitch` for a switch with a numeric label.
+2. `ByteInput` which contains 8 `BitSwitch` widgets.
+3. `ByteEditor` which contains a `ByteInput` and an [Input](../widgets/input.md) to show the decimal value.
+
+This is not the only way we could implement our design with compound widgets.
+So why these three widgets?
+As a rule of thumb, a widget should handle one piece of data, which is why we have an independent widget for a bit, a byte, and the decimal value.
+
+<div class="excalidraw">
+--8<-- "docs/images/byte02.excalidraw.svg"
+</div>
+
+In the following code we will implement the three widgets. There will be no functionality yet, but it should look like our design.
+
+=== "byte01.py"
+
+    ```python title="byte01.py" hl_lines="28-30 48-50 67-71"
+    --8<-- "docs/examples/guide/compound/byte01.py"
+    ```
+
+=== "Output"
+
+    ```{.textual path="docs/examples/guide/compound/byte01.py" columns="90" line="30"}
+    ```
+
+Note the `compose()` methods of each of the widgets.
+
+- The `BitSwitch` yields a [Label](../widgets/label.md) which displays the bit number, and a [Switch](../widgets/switch.md) control for that bit. The default CSS for `BitSwitch` aligns its children vertically, and sets the label's [text-align](../styles/text_align.md) to center.
+
+- The `ByteInput` yields 8 `BitSwitch` widgets and arranges them horizontally. It also adds a `focus-within` style in its CSS to draw an accent border when any of the switches are focused.
+
+- The `ByteEditor` yields a `ByteInput` and an Input control. The default CSS stacks the two controls on top of each other to divide the screen in to two parts.
+
+With these three widgets, the [DOM](CSS.md#the-dom) for our app will look like this:
+
+<div class="excalidraw">
+--8<-- "docs/images/byte_input_dom.excalidraw.svg"
```

Comment:
Not sure if it was deliberate or just accidental, but how do we stand on using informal language like "thru" in the docs?

Reply:
I think it works in the diagrams, which have an informal look to them. We probably shouldn't use it in the body of the text tho.